packages/grid/_modules_/grid/hooks/features/rows/useRows.ts

```diff
@@ -88,11 +91,14 @@ export const useRows = (rows: RowsProp, apiRef: ApiRef): void => {
           ? gridState.options.rowCount
           : allRows.length;
 
-      internalRowsState.current = { idRowsLookup, allRows, totalRowCount };
+      internalRowsState.current = {idRowsLookup, allRows, totalRowCount};
+
+      setGridState((state) => ({...state, rows: internalRowsState.current}));
+
+      apiRef.current.applySorting(true);
+      apiRef.current.applyFilters(true);
+      forceUpdate(() => apiRef.current.publishEvent(ROWS_UPDATED));
```

Comment:
Consider adding a reset row events to avoid calling applySorting directly

Reply:
Do we need the returned value? It's not obvious. Isn't `forceUpdate` a `setState` behind the scene (so the returned value matter)? 

```suggestion
      forceUpdate(() => { apiRef.current.publishEvent(ROWS_UPDATED); });
```

Reply:
No the returned value does not matter, as the forceUpdate is just to trigger a render, it does not update anything. The state update is always done in SetGridState.
the return value is not used, not adding the {} is for code clarity and do a one-liner function instead of 3.

Reply:
> the return value is not used, not adding the {} is for code clarity and do a one-liner function instead of 3.

On the other hand, it removes any potential source of confusion. In `useGridApi.ts`, it matters, it's done like this:

`const [, forceUpdate] = React.useState<GridState>();`, it only rerender if force update is called with a new state that is referential different from the previous one.